sdk/formrecognizer/Azure.AI.FormRecognizer/src/StreamExtensions.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using Azure.AI.FormRecognizer.Models;
+
+namespace Azure.AI.FormRecognizer
+{
+    internal static class StreamExtensions
+    {
+        /// <summary>The set of bytes expected to be present at the start of PDF files.</summary>
+        private static byte[] PdfHeader = new byte[] { 0x25, 0x50, 0x44, 0x46 };
+
+        /// <summary>The set of bytes expected to be present at the start of PNG files.</summary>
+        private static byte[] PngHeader = new byte[] { 0x89, 0x50, 0x4E, 0x47 };
+
+        /// <summary>The set of bytes expected to be present at the start of JPEG files.</summary>
+        private static byte[] JpegHeader = new byte[] { 0xFF, 0xD8 };
+
+        /// <summary>The set of bytes expected to be present at the start of TIFF (little-endian) files.</summary>
+        private static byte[] TiffLeHeader = new byte[] { 0x49, 0x49, 0x2A, 0x00 };
+
+        /// <summary>The set of bytes expected to be present at the start of TIFF (big-endian) files.</summary>
+        private static byte[] TiffBeHeader = new byte[] { 0x4D, 0x4D, 0x00, 0x2A };
+
+        /// <summary>
+        /// Attemps to detect the <see cref="ContentType"/> of a stream of bytes. The algorithm searches through
+        /// the first set of bytes in the stream and compares it to well-known file signatures.
+        /// </summary>
+        /// <param name="stream">The stream to which the content type detection attempt will be performed.</param>
+        /// <param name="contentType">If the detection is successful, outputs the detected content type. Otherwise, <c>default</c>.</param>
+        /// <returns><c>true</c> if the detection was successful. Otherwise, <c>false</c>.</returns>
+        /// <exception cref="NotSupportedException">Happens when <paramref name="stream"/> is not seekable or readable.</exception>
```

Comment:
Just curious ... why throw NotSupportedException here when ArgumentException is thrown above for the same error case?  Would it make sense for these to be consistent?

Reply:
`Azure.Core.RequestContent.StreamContent` throws `ArgumentException`:
https://github.com/Azure/azure-sdk-for-net/blob/73551953eb31c81f66c44970a46c7f0d1005067a/sdk/core/Azure.Core/src/RequestContent.cs#L91

Reply:
`NotSupportedException` is thrown by the stream itself when the seek or read operations are not supported and this extension method is called. This is not something we can change. I added a comment about the `NotSupportedException` here in case someone tries to use this method somewhere else in the future, but it's not an error we expect to trigger.

In the client, we manually check if the stream is seekable and readable before calling `TryGetContentType`. We throw an `ArgumentException` in case of failure because the set of arguments specified by the user do not have enough information for us to make a valid request (no content type specified in the options + unreadable/unseekable stream), so it seems a problem with the arguments.